modules/eks/cluster/addons.tf

```
@@ -0,0 +1,76 @@
+# https://docs.aws.amazon.com/eks/latest/userguide/eks-add-ons.html
+# https://docs.aws.amazon.com/eks/latest/userguide/managing-add-ons.html#creating-an-add-on
+
+locals {
+  eks_cluster_oidc_issuer_url = replace(local.eks_outputs.eks_cluster_identity_oidc_issuer, "https://", "")
+
+  addon_names                = keys(var.addons)
+  vpc_cni_addon_enabled      = local.enabled && contains(local.addon_names, "vpc-cni")
+  aws_ebs_csi_driver_enabled = local.enabled && contains(local.addon_names, "aws-ebs-csi-driver")
+
+  # The `vpc-cni` and `aws-ebs-csi-driver` addons are special as they always require an IAM role for Kubernetes Service Account (IRSA).
+  # The roles are created by this component.
+  addon_service_account_role_arn_map = {
+    vpc-cni            = module.vpc_cni_eks_iam_role.service_account_role_arn
+    aws-ebs-csi-driver = module.aws_ebs_csi_driver_eks_iam_role.service_account_role_arn
+  }
+
+  addons = [
+    for k, v in var.addons : {
+      addon_name               = k
+      addon_version            = lookup(v, "addon_version", null)
+      resolve_conflicts        = lookup(v, "resolve_conflicts", null)
+      service_account_role_arn = try(coalesce(lookup(v, "service_account_role_arn", null), lookup(local.addon_service_account_role_arn_map, k, null)), null)
+    }
+  ]
+}
+
+# https://docs.aws.amazon.com/eks/latest/userguide/cni-iam-role.html
+# https://docs.aws.amazon.com/eks/latest/userguide/managing-vpc-cni.html
+# https://docs.aws.amazon.com/eks/latest/userguide/cni-iam-role.html#cni-iam-role-create-role
+# https://aws.github.io/aws-eks-best-practices/networking/vpc-cni/#deploy-vpc-cni-managed-add-on
+data "aws_iam_policy" "vpc_cni" {
+  count = local.vpc_cni_addon_enabled ? 1 : 0
+  arn   = "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy"
+}
```

Attach AWS managed policies directly, so that we get updates when AWS updates the policies.

```suggestion
resource "aws_iam_role_policy_attachment" "vpc_cni" {
  count = local.vpc_cni_addon_enabled ? 1 : 0

  role       = module.vpc_cni_eks_iam_role.service_account_role_arn
  policy_arn = "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy"
}
```